guide/interactions/slash-command-permissions.md

```diff
@@ -1,104 +1,221 @@
 
-await command.permissions.add({ permissions });
+The **allowlist** approach does the opposite. It DISABLES the command for everyone (i.e. `setDefaultPermission(false)`), and then manually ENABLE the command for a specific **user** or **role** in a specific guild.
+
+The process of "manually enabling/disabling the command for a specific user/role" is called **permission overwrite**.
+
+::: warning
+You can only add up to **10 permission overwrites** for the same command.
+
+**Permission overwrites are guild-specific**. In other words, you need to specify the guild that you want to do the permission overwrite in, **even if the command is a Global command**. That means that, for a given Global command, it's possible to enable it for a user in one guild, but disable it for the same user in another guild.
+:::
+
+Let's try the allowlist approach for the `ping` command. First, you set the `default_permission` to `false` by using the `setDefaultPermission()` method:
+```js:no-line-numbers {4}
+new SlashCommandBuilder()
+  .setName('ping')
+  .setDescription('Replies with pong!')
+  .setDefaultPermission(false)
 ```
 
-Now you have successfully denied the user whose `id` you used access to this application command.
+Next, you run the deployment script again to register your newly edited `ping` command:
+```sh:no-line-numbers
+node deploy-commands.js
+```
 
-::: tip
-If you want to update permissions for a global command instead, your `command` variable would be:
-```js
-const command = client.application?.commands.fetch('123456789012345678');
+Go ahead and try using the `/ping` command in your guild. You should see it grayed out and disabled like this:
+
+![](./images/disabled-command.png)
+
+Next, let's enable the `ping` command for a specific role. You need three IDs for a permission overwrite:
+
+- Guild ID
+- Command ID
+- User/role ID
+
+So far, you already have the Guild ID in the `deploy-commands.js` file.
+
+- Guild ID ✅
+- Command ID
+- User/role ID
+
+With these three IDs, you make another HTTP PUT request but to a different endpoint (route). As you might've guessed, that means we need to go back to your `deploy-commands.js` file to implement a few more lines of code:
+
+:::: code-group
+::: code-group-item deploy-commands.js
+<!-- eslint-skip -->
+```js {7,16,23,25-36,38-41,43}
+const { SlashCommandBuilder } = require('@discordjs/builders');
+const { REST } = require('@discordjs/rest');
+const { Routes } = require('discord-api-types/v9');
+const { clientId, guildId, token } = require('./config.json');
+
+const commands = [
+	new SlashCommandBuilder().setName('ping').setDescription('Replies with pong!').setDefaultPermission(false),
+	new SlashCommandBuilder().setName('server').setDescription('Replies with server info!'),
+	new SlashCommandBuilder().setName('user').setDescription('Replies with user info!'),
+].map(command => command.toJSON());
```

Comment:
Just to make sure I understood what you meant here: you're suggesting replacing the entire array with a `// ...`?

Would you mind elaborating on the reason behind this suggestion?

Reply:
The guide, in order to reduce redundancy, uses `// ...` whenever a chunk of code is the same as earlier. Moreover, This `commands` constant can differ based on the user and what they have in their file so it's better to use this, imo.

Lmk what you think.

Reply:
While i agree with reducing redundancy with something like `// ...`, i ultimately decided to utilize line highlighting to show the relevant piece of code. I know the guide expects the reader to have basic knowledge of programming and JS, but that's not the case in reality (unfortunately). As a result, I believe showing the complete code block is more "beginner friendly".

I would love to refactor all the code blocks i've written to use something like `// ...`, but i'm afraid this might cause confusion for majority of people reading the guide.

If we assume that the reader is able to "fill in the gap" when looking at something like `// ...`, then I think it's also safe to assume that the reader can replace `const commands = [ new SlashCommandBuilder()... ]` with their own code.

Reply:
Hmm, true.

Maybe:
```suggestion
const commands = [
  // ...
];
```

Reply:
Alright. So, I guess I should also remove `const { SlashCommandBuilder } = require('@discordjs/builders');`, correct?